### PR TITLE
Sc2 fixes

### PR DESCRIPTION
**Pull request by yger**
MInor improvements for SC2

### CODE REVIEW

**Comment by yger** [comment]
This fixes the reload bug in SC2. @samuelgarcia could you approve this such that I can merge, and get a stable behavior for SC2?

**Comment by yger** [comment]
In fact no, I will need your magic   input @samuelgarcia, there is something weird with the classmethods

**Comment by yger** [comment]
This has been fixed! Good to merge

File: src/spikeinterface/sortingcomponents/clustering/clustering_tools.py [hunk 1]
```diff
@@ -593,29 +593,34 @@ def remove_duplicates_via_matching(
...
     )
 
+    spikes_per_units, counts = np.unique(waveform_extractor.sorting.to_spike_vector()["unit_index"], return_counts=True)
+    indices = np.argsort(counts)
+
     ignore_ids = []
     similar_templates = [[], []]
 
-    for i in range(nb_templates):
+    for i in np.arange(nb_templates)[indices]:
```
Comment (samuelgarcia) [inline src/spikeinterface/sortingcomponents/clustering/clustering_tools.py:613]: My guess is that 
 `np.arange(nb_templates)[indices]`is the same as `indices`no ?

File: src/spikeinterface/sortingcomponents/clustering/random_projections.py [hunk 2]
```diff
@@ -223,6 +223,9 @@ def sigmoid(x, L, x0, k, b):
         )
 
         del we, sorting
+        import gc
+
+        gc.collect()
```
Comment (samuelgarcia) [inline src/spikeinterface/sortingcomponents/clustering/random_projections.py:228]: No manual gc hidden in the code please!

**Review by samuelgarcia — APPROVED** [review]

**Pull request merged**